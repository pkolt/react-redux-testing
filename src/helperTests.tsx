import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore, { AppState } from './store';

interface WrappingComponentProps {
  children?: React.ReactNode;
  storeState?: AppState;
  historyState?: string[];
}

export const WrappingComponent: React.FC<WrappingComponentProps> = (props: WrappingComponentProps) => {
  const { children, storeState, historyState } = props;
  const store = configureStore(storeState);

  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={historyState} keyLength={0}>
        {children}
      </MemoryRouter>
    </Provider>
  );
};
