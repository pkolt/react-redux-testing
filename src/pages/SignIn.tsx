import React from 'react';
import { useForm } from 'react-hook-form';
import classNames from 'classnames';
import Button from '../components/Button';
import { useDispatch } from 'react-redux';
import { userActions } from '../store/user';
import { useHistory } from 'react-router-dom';
import { URLS } from '../constants';

const SignIn: React.FC = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const { register, handleSubmit, errors } = useForm();
  const onSubmit = (data: Record<string, string>) => {
    const {login, password} = data;
    if (password === '12345') {
        dispatch(userActions.changedData({login, password}));
        history.replace(URLS.DASHBOARD);
    }
  };
  const loginClsName = classNames('form-control', { 'is-invalid': errors.login });
  const passwordClsName = classNames('form-control', { 'is-invalid': errors.password });

  return (
    <div className="bg-light d-flex h-100 align-items-center justify-content-center">
      <div className="card w-25">
        <h4 className="card-header text-center">Авторизация</h4>
        <div className="card-body">
          <form onSubmit={handleSubmit(onSubmit)}>
            <div className="form-group">
              <label htmlFor="login">Пользователь</label>
              <input
                type="text"
                name="login"
                id="login"
                className={loginClsName}
                ref={register({ required: true })}
              />
              <div className="invalid-feedback">Обязательное поле</div>
            </div>
            <div className="form-group">
              <label htmlFor="password">Пароль</label>
              <input
                type="password"
                name="password"
                id="password"
                className={passwordClsName}
                ref={register({ required: true })}
              />
              <div className="invalid-feedback">Обязательное поле</div>
            </div>
            <div className="d-flex justify-content-center">
              <Button modifier="btn-primary">Войти</Button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default SignIn;
