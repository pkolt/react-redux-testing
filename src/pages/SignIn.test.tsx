import React from 'react';
import SignIn from './SignIn';
import { mount } from 'enzyme';
import { WrappingComponent } from '../helperTests';

describe('<SignIn />', () => {
  it('renders without crashing', () => {
    const wrapper = mount(<SignIn />, {
        wrappingComponent: WrappingComponent,
    });
    expect(wrapper).toMatchSnapshot();
  });
});