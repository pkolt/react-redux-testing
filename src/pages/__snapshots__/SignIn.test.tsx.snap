// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> renders without crashing 1`] = `
<SignIn>
  <div
    className="bg-light d-flex h-100 align-items-center justify-content-center"
  >
    <div
      className="card w-25"
    >
      <h4
        className="card-header text-center"
      >
        Авторизация
      </h4>
      <div
        className="card-body"
      >
        <SignInForm
          onSubmit={[Function]}
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="login"
              >
                Пользователь
              </label>
              <input
                className="form-control"
                id="login"
                name="login"
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="password"
              >
                Пароль
              </label>
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[Function]}
                type="password"
              />
            </div>
            <div
              className="d-flex justify-content-center"
            >
              <Button
                modifier="btn-primary"
              >
                <button
                  className="btn btn-primary"
                  type="submit"
                >
                  Войти
                </button>
              </Button>
            </div>
          </form>
        </SignInForm>
      </div>
    </div>
  </div>
</SignIn>
`;
